### PR TITLE
[pre-commit MyPy] Align the entire repo with MyPy #3

### PR DESCRIPTION
**Pull request by mmhw**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
Related: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-8236)

## Description
In order to run `pre-commit` on our entire repo, it is necessary to fix certain files that fail the `mypy` hook.

### CODE REVIEW

File: Packs/AWS-GuardDuty/Integrations/AWSGuardDuty/AWSGuardDuty.py
Comment (shmuel44) [inline Packs/AWS-GuardDuty/Integrations/AWSGuardDuty/AWSGuardDuty.py:1]: line 775, add type hint
 `client: "GuardDutyClient" = aws_client.aws_session(...`

File: Packs/AWS-GuardDuty/Integrations/AWSGuardDutyEventCollector/AWSGuardDutyEventCollector.py
Comment (shmuel44) [inline Packs/AWS-GuardDuty/Integrations/AWSGuardDutyEventCollector/AWSGuardDutyEventCollector.py:1]: line 193, add hint type
client: "GuardDutyClient" = aws_client.aws_session(...

File: Packs/AWS-SecurityHub/Integrations/AWSSecurityHubEventCollector/AWSSecurityHubEventCollector.py
Comment (shmuel44) [inline Packs/AWS-SecurityHub/Integrations/AWSSecurityHubEventCollector/AWSSecurityHubEventCollector.py:1]: line 244, add hint type
`client: "SecurityHubClient" = aws_client.aws_session(...`

File: Packs/AWS_WAF/Integrations/AWSWAF/AWSWAF.py
Comment (shmuel44) [inline Packs/AWS_WAF/Integrations/AWSWAF/AWSWAF.py:1]: line 1055, add hint type
`client: "WAFV2Client" = aws_client.aws_session(...`

File: Packs/AWS-GuardDuty/Integrations/AWSGuardDuty/AWSGuardDuty.py [hunk 1]
```diff
@@ -429,13 +430,15 @@ def get_pagination_args(args: dict) -> Tuple[Optional[int], Optional[int], Optio
 
     # Automatic Pagination
     limit = arg_to_number(args.get('limit', MAX_RESULTS_RESPONSE))
+    assert isinstance(limit, int)
```
Comment (dorschw) [inline Packs/AWS-GuardDuty/Integrations/AWSGuardDuty/AWSGuardDuty.py:433]:
```suggestion
    assert isinstance(limit, int) # for mypy
```

File: Packs/AWS-GuardDuty/Integrations/AWSGuardDuty/AWSGuardDuty.py [hunk 2]
```diff
@@ -429,13 +430,15 @@ def get_pagination_args(args: dict) -> Tuple[Optional[int], Optional[int], Optio
...
 
     # Manual Pagination
     page = arg_to_number(args.get('page'))
     if page is not None and page <= 0:
         raise DemistoException('page argument must be greater than 0')
 
     page_size = arg_to_number(args.get('page_size', MAX_RESULTS_RESPONSE))
+    assert isinstance(page_size, int)
```
Comment (dorschw) [inline Packs/AWS-GuardDuty/Integrations/AWSGuardDuty/AWSGuardDuty.py:441]:
```suggestion
    assert isinstance(page_size, int) # for mypy
```

**Review by dorschw — APPROVED** [review]

**Review by dorschw — APPROVED** [review]

**Comment by mmhw** [comment]
Reason for force merge:
[run-unittests-and-lint](https://code.pan.run/xsoar/content/-/jobs/33658048) - Unit-tests coverage.

**Pull request merged**